analyzers/nodejs/integration_test.go

```diff
@@ -0,0 +1,74 @@
+// +build integration
+
+package nodejs_test
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/fossas/fossa-cli/analyzers"
+	"github.com/fossas/fossa-cli/files"
+	"github.com/fossas/fossa-cli/module"
+	"github.com/fossas/fossa-cli/pkg"
+	"github.com/fossas/fossa-cli/test/project_initializer/node"
+	"github.com/fossas/fossa-cli/test/testtools"
+	"github.com/stretchr/testify/assert"
+)
+
+var nodeInitializer = node.New()
+
+func TestMain(m *testing.M) {
+	teardownTests, err := testtools.SetupTests(nodeInitializer)
+	if err != nil {
+		if err.Error() != "repository already exists" {
+			panic(err)
+		}
+	}
+
+	exitCode := m.Run()
+	defer os.Exit(exitCode)
+	defer teardownTests()
+}
+
+// While not testing the core functionality, this ensures that the tests have been setup correctly as needed for a prereq to run the analyzer steps
+// This test itself does not incur any overhead.
+func TestTestSetup(t *testing.T) {
+	assertProjectFixtureExists(t, "puppeteer")
+	// faker has no deps
+	// assertProjectFixtureExists(t, "fakerjs")
+	assertProjectFixtureExists(t, "fastify")
+	assertProjectFixtureExists(t, "nest")
+	assertProjectFixtureExists(t, "ohm")
+	assertProjectFixtureExists(t, "express")
+	assertProjectFixtureExists(t, "standard")
+	assertProjectFixtureExists(t, "sodium-encryption")
+	assertProjectFixtureExists(t, "request")
+}
+
+func TestStandardJsAnalysis(t *testing.T) {
+	module := module.Module{
+		Dir:     filepath.Join(nodeInitializer.FixtureDirectory(), "standard"),
+		Type:    pkg.NodeJS,
+		Name:    "standardjs",
+		Options: map[string]interface{}{"allow-npm-err": true},
+	}
+
+	analyzer, err := analyzers.New(module)
+	assert.NoError(t, err)
+
+	deps, err := analyzer.Analyze()
+	assert.NoError(t, err)
+	assert.NotEmpty(t, deps.Direct)
+	assert.NotEmpty(t, deps.Transitive)
```

Comment:
While these are not in depth, they yield the same level of confidence that `test.sh` would give